pandas/_libs/tslibs/period.pyx

```diff
@@ -2664,3 +2677,28 @@ def validate_end_alias(how: str) -> str:  # Literal["E", "S"]
     if how not in {"S", "E"}:
         raise ValueError("How must be one of S or E")
     return how
+
+
+cdef _parse_weekly_str(value, BaseOffset freq):
+    """
+    Parse e.g. "2017-01-23/2017-01-29", which cannot be parsed by the general
+    datetime-parsing logic.  This ensures that we can round-trip with
+    Period.__str__ with weekly freq.
+    """
+    # GH#50803
+    start, end = value.split("/")
+    start = Timestamp(start)
```

Comment:
Any reason to use Timestamp here? 

IIUC from the last time I was in Period code, Timestamp sends strings down ``parse_datetime_string_with_reso``.

Reply:
could use datetime.strptime.  either way is fine.  Timestamp doesn't _yet_ use parse_datetime_string_with_reso, but getting there is one of the goals this is aiming at.  even when it does, that won't affect behavior here.